download.go

```diff
@@ -0,0 +1,314 @@
+	if err := MakeEnclosingDirectories(opts.Path); err != nil {
+		return errors.Wrap(err, "problem creating enclosing directories")
+	}
+
+	feed, err := bond.GetArtifactsFeed(opts.Path)
+	if err != nil {
+		return errors.Wrap(err, "problem making artifacts feed")
+	}
+
+	catcher := grip.NewBasicCatcher()
+	urls, errs1 := feed.GetArchives(opts.Releases, opts.BuildOpts)
+	jobs, errs2 := createDownloadJobs(opts.Path, urls)
+
+	if err := setupDownloadJobsAsync(ctx, jobs, processDownloadJobs(ctx, addMongoDBFilesToCache(cache, opts.Path))); err != nil {
+		catcher.Add(errors.Wrap(err, "problem starting download jobs"))
+	}
+
+	if err := aggregateErrors(errs1, errs2); err != nil {
+		catcher.Add(errors.Wrap(err, "problem initializing download jobs"))
+	}
+
+	return catcher.Resolve()
+}
+
+func createDownloadJobs(path string, urls <-chan string) (<-chan amboy.Job, <-chan error) {
+	output := make(chan amboy.Job)
+	errOut := make(chan error)
+
+	go func() {
+		catcher := grip.NewCatcher()
+		for url := range urls {
+			j, err := recall.NewDownloadJob(url, path, true)
+			if err != nil {
+				catcher.Add(errors.Wrapf(err, "problem creating download job for %s", url))
+				continue
+			}
+
+			output <- j
+		}
+		close(output)
+		if catcher.HasErrors() {
+			errOut <- catcher.Resolve()
+		}
+		close(errOut)
+	}()
+
+	return output, errOut
+}
+
+func aggregateErrors(groups ...<-chan error) error {
+	catcher := grip.NewCatcher()
+
+	for _, g := range groups {
+		for err := range g {
+			catcher.Add(err)
+		}
+	}
+
+	return catcher.Resolve()
+}
+
+func processDownloadJobs(ctx context.Context, processFile func(string) error) func(amboy.Queue) error {
+	return func(q amboy.Queue) error {
+		grip.Infof("waiting for %d download jobs to complete", q.Stats().Total)
+		_ = amboy.WaitCtxInterval(ctx, q, 1000*time.Millisecond)
+		grip.Info("all download tasks complete, processing errors now")
+
+		if err := amboy.ResolveErrors(ctx, q); err != nil {
+			return errors.Wrap(err, "problem completing download jobs")
+		}
+
+		catcher := grip.NewBasicCatcher()
+		for job := range q.Results(ctx) {
+			downloadJob, ok := job.(*recall.DownloadFileJob)
+			if !ok {
+				catcher.Add(errors.New("problem retrieving download job from queue"))
+				continue
+			}
+			if err := processFile(downloadJob.FileName); err != nil {
+				catcher.Add(err)
+			}
+		}
+		return catcher.Resolve()
+	}
+}
+
+func setupDownloadJobsAsync(ctx context.Context, jobs <-chan amboy.Job, processJobs func(amboy.Queue) error) error {
+	q := queue.NewLocalUnordered(runtime.NumCPU())
+	if err := q.Start(ctx); err != nil {
+		return errors.Wrap(err, "problem starting download job queue")
+	}
+
+	if err := amboy.PopulateQueue(ctx, q, jobs); err != nil {
+		return errors.Wrap(err, "problem adding download jobs to queue")
+	}
+
+	go func() {
+		if err := processJobs(q); err != nil {
+			grip.Errorf(errors.Wrap(err, "error occurred while adding jobs to cache").Error())
```

Comment:
what do we do this this error?

Reply:
Truthfully, not much. It's only there to print a message in case an error occurs while it's asynchronously downloading.